features/netsocket/emac-drivers/TARGET_WHD/interface/whd_emac.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "cmsis_os.h"
+#include "whd_emac.h"
+#include "lwip/etharp.h"
+#include "lwip/ethip6.h"
+#include "mbed_shared_queues.h"
+#include "whd_wlioctl.h"
+#include "whd_buffer_api.h"
+#include "cybsp_api_wifi.h"
+#include "emac_eapol.h"
+#include "cy_result.h"
+
+extern "C"
+{
+    eapol_packet_handler_t emac_eapol_packet_handler = NULL;
+} // extern "C"
+
+WHD_EMAC::WHD_EMAC(whd_interface_role_t role)
+    : interface_type(role)
+{
+}
+
+WHD_EMAC::WHD_EMAC()
+    : interface_type(WHD_STA_ROLE)
+{
+}
+
+WHD_EMAC &WHD_EMAC::get_instance() {
```

Comment:
this file should follow Mbed OS style (please run astyle)

Reply:
Re-applied astyle: https://github.com/ARMmbed/mbed-os/pull/10692/commits/85a5bc0bf9eeac3565a933c46d1363825feb8d39